### PR TITLE
perf(logits): drop redundant nonzero host-device sync in chunked input-logprob path

### PR DESCRIPTION
**Pull request by puneethgv**
## Motivation

Addresses part of #9889 (remove `torch.nonzero()` where it forces a host-device sync).

In `LogitsProcessor.process_input_logprobs_by_chunk`, every chunk iteration executes:

```python
mask_indices = torch.nonzero(chunk_mask, as_tuple=True)[0]
```

solely so it can gather `extend_input_logprob_token_ids_gpu[mask_indices]`. When `chunk_mask` is on CUDA, `torch.nonzero` triggers a **host-device synchronization** — its output length is data-dependent, so the GPU must drain and report the count back to the host, stalling the pipeline on the return-logprob prefill path.

## Modifications

For a 1-D boolean mask, `x[mask.nonzero(as_tuple=True)[0]]` is exactly equivalent to direct boolean indexing `x[mask]`. `extend_input_logprob_token_ids_gpu` is length-N in the same position space as `chunk_mask` (it is indexed 1:1 with the input-logprob rows in the non-chunked path, and built at `forward_batch_info.py`), so we can index it directly with `chunk_mask` and drop the intermediate `mask_indices` tensor entirely.

- Remove the `mask_indices = torch.nonzero(chunk_mask, as_tuple=True)[0]` line.
- Index `extend_input_logprob_token_ids_gpu[chunk_mask]` directly.

This removes one host-device sync (and one temporary tensor) per chunk iteration. The output is **byte-identical** — this is a pure, output-preserving micro-optimization.

Net diff: `+1 / -4` in `python/sglang/srt/layers/logits_processor.py`.

## Accuracy Tests

Output is unchanged by construction (the two indexing forms are provably equivalent for a 1-D boolean mask). The existing input-logprob tests exercise this path and should remain green:

- `test/registered/sampling/test_original_logprobs.py`
- `test/manual/test_logprobs.py`

The chunked branch is reached with chunked prefill enabled and a prompt longer than the chunk size with `return_logprob=True` (e.g. a small `--chunked-prefill-size`).

## Speed Tests and Profiling

Micro-optimization on the return-logprob prefill path: eliminates one `nonzero`-induced host-device sync per chunk iteration. No change to the steady-state decode path.

## Checklist

- [x] Format your code according to the Format code with pre-commit.
- [x] Add unit tests according to the Run and add unit tests. (No new test — output is provably identical; covered by existing logprob tests above.)
- [ ] Update documentation according to Write documentations. (N/A — no user-facing behavior change.)
- [ ] Provide accuracy and speed benchmark results. (N/A — output-preserving refactor.)
- [x] Follow the SGLang code style guidance.

---
### CI States

Latest PR Test (Base): :x: [Run #28668148269](https://github.com/sgl-project/sglang/actions/runs/28668148269)
Latest PR Test (Extra): :x: [Run #28668148117](https://github.com/sgl-project/sglang/actions/runs/28668148117)

### CODE REVIEW

**Comment by puneethgv** [comment]
Closing this one. On a closer look the change is net-neutral: dropping the explicit `torch.nonzero(chunk_mask)` and gathering with `extend_input_logprob_token_ids_gpu[chunk_mask]` just moves the equivalent host-device sync into the boolean index, so it doesn't actually remove a sync. It also overlaps #25508, which already refactors this same `process_input_logprobs_by_chunk` path. Not worth the review churn — withdrawing. Thanks!
